lib/index.js

```diff
@@ -0,0 +1,240 @@
+    // allow up to 10x inefficiency before trying to recover
+    recoveryGenerationThreshold = minimumMergeEvents * 10;
+  }
+
+  logger.verbose(
+    'Selected Electors:',
+    {ledgerNode: ledgerNode.id, blockHeight, electors, recoveryElectors});
+
+  return {
+    // number of electors, must be 3f+1 (or 1 for dictator model), safe
+    // for up to `f` byzantine failures
+    electors,
+    // optional `f+1` "trusted" electors that must not fail and can handle
+    // catastrophic case where remaining `2f` electors cannot communicate
+    // (recovery electors are not a failsafe against a coordinated byzantine
+    // attack as the byzantine electors could collude to show different sides
+    // of a fork, but safe against network failures preventing consensus)
+    recoveryElectors,
+    // number of generations of merge events where no new electors participate
+    // and fewer than 2f+1 total electors (including self) participate causing
+    // recovery mode to trigger because consensus cannot be reached
+    recoveryGenerationThreshold,
+  };
+};
+
+// it is useful to override this function in tests
+api._computeElectors = async ({
+  blockHeight, electorPoolElectors, latestBlockSummary, ledgerConfiguration,
+  ledgerNode
+}) => {
+  // get partipicants for the last block
+
+  // FIXME: if we're using the _blocks API here, should it be made public
+  // (remove the underscore?)
+
+  const {_blocks} = ledgerNode.consensus;
+  const {consensusProofPeers, mergeEventPeers} = await _blocks.getParticipants(
+    {blockHeight: blockHeight - 1, ledgerNode});
+
+  // TODO: we should be able to easily remove previously detected
+  // byzantine nodes (e.g. those that forked at least) from the electors
+
+  // TODO: simply count consensus event signers once and proof signers
+  //   twice for now -- add comprehensive elector selection and
+  //   recommended elector vote aggregating algorithm in v2
+  const aggregate = {};
+  // add merge event peers
+  for(const id of mergeEventPeers) {
+    aggregate[id] = {id, weight: 1};
+  }
+
+  // TODO: weight previous electors more heavily to encourage continuity
+  // add consensusProof peers
+  for(const id of consensusProofPeers) {
+    if(id in aggregate) {
+      aggregate[id].weight = 3;
+    } else {
+      aggregate[id] = {id, weight: 2};
+    }
+  }
+  // add peers from the electorPool
+  for(const elector of electorPoolElectors) {
+    const electorType = [].concat(elector.type);
+    // electors defined in *any* way in the elector pool document should
+    // have the greatest weight
+    // FIXME: weight these like this?
+    // FIXME: is type === Continuity2017Peer a real thing?  If so, what do we
+    // do for that.
+    let weight = 100;
+    let recoveryElector = false;
+    if(electorType.includes(ELECTOR_TYPE_RECOVERY)) {
+      weight += 100;
+      recoveryElector = true;
+    }
+    if(electorType.includes(ELECTOR_TYPE_GUARANTOR)) {
+      weight += 100;
+    }
```

Comment:
I haven't considered if this weighting formula works ... and it seems like an indirect way to just do exactly what we want. I recommend we remove it and be more clear in the algorithm about the intent. We'll need to add those other types to the context as well.